internal/provider/resource_tag.go

```diff
@@ -362,3 +362,29 @@ func setTagAttributes(d *schema.ResourceData, c *SchemaRegistryRestClient, clust
 	d.SetId(createTagId(clusterId, tag.GetName()))
 	return d, nil
 }
+
+func extractCatalogRestEndpoint(client *Client, d *schema.ResourceData, isImportOperation bool) (string, error) {
+	if client.isSchemaRegistryMetadataSet {
+		if client.catalogRestEndpoint != "" {
+			return client.catalogRestEndpoint, nil
+		} else {
+			return client.schemaRegistryRestEndpoint, nil
+		}
+	}
```

Comment:
nit: did we try to see how this could would look like if we fully decouple these clients? Long term catalog client may have separate set of API Keys, so we'll have to refactor it from SR client completely.